lib/transformer/README.md

```diff
@@ -0,0 +1,94 @@
+# AST Transformer
+
+After parsing the files imported by the entry files, the Abstract Syntax Tree (AST) is made.  With the `--transform` argument, the compiler will load any js files provided and if the exports include a Transformer class.
+
+## Transformer Class
+
+By extending the Transformer class your subclass gains access to the `Parser`, `writeFile` function, `baseDir` is where the project's root is located, and `stdout` and `stderr`
+
+```ts
+export abstract class Transformer {
+  constructor(
+    protected parser: Parser,
+    protected writeFile: FileWriter,
+    protected baseDir: string,
+    protected stdout: (data: string) => void,
+    protected stderr: (data: string) => void
+  ) {}
+
+  get program(): Program {
+    return this.parser.program;
+  }
+
+  abstract afterParse(): void;
+}
+```
+
+### Example ASTPrinter
+
+Below is an example of visiting each source that is a user entry file and creating a string from visiting the AST.
+
+```ts
+/**
+ * Example of using a transformer to print the AST for each entry file
+ */
+export default class Printer extends Transformer {
```

Comment:
According to latest best of practice for JS/TS we should [avoid](https://blog.neufund.org/why-we-have-banned-default-exports-and-you-should-do-the-same-d51fdc2cf2ad) [`default`](https://humanwhocodes.com/blog/2019/01/stop-using-default-exports-javascript-module/) [exports](https://basarat.gitbooks.io/typescript/docs/tips/defaultIsBad.html)